providers/cloudns/cloudnsProvider.go

```diff
@@ -214,13 +219,24 @@ func toRc(domain string, r *domainRecord) *models.RecordConfig {
 		sshfpFingerprint, _ := strconv.ParseUint(r.SshfpFingerprint, 10, 32)
 		rc.SshfpFingerprint = uint8(sshfpFingerprint)
 		rc.SetTarget(r.Target)
+	case "DS":
+		dsKeyTag, _ := strconv.ParseUint(r.DsKeyTag, 10, 32)
+		rc.DsKeyTag = uint16(dsKeyTag)
+		dsAlgorithm, _ := strconv.ParseUint(r.SshfpAlgorithm, 10, 32)
+		rc.DsAlgorithm = uint8(dsAlgorithm)
+		dsDigestType, _ := strconv.ParseUint(r.DsDigestType, 10, 32)
+		rc.DsDigestType = uint8(dsDigestType)
+		rc.DsDigest = r.Target
+		rc.SetTarget(r.Target)
+
 	default:
 		rc.SetTarget(r.Target)
 	}
 
 	return rc
 }
 
+//generates the update requests sent to the API (it takes a RecordConfig and turns it into the request).
```

Comment:
Suggested comment:

// toReq takes a RecordConfig and turns it into the native format used by the API.